src/main/java/org/codehaus/mojo/flatten/DirectDependenciesInheritanceAssembler.java

```diff
@@ -0,0 +1,158 @@
+/*
+  * Licensed to the Apache Software Foundation (ASF) under one
+  * or more contributor license agreements.  See the NOTICE file
+  * distributed with this work for additional information
+  * regarding copyright ownership.  The ASF licenses this file
+  * to you under the Apache License, Version 2.0 (the
+  * "License"); you may not use this file except in compliance
+  * with the License.  You may obtain a copy of the License at
+  *
+  *   http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing,
+  * software distributed under the License is distributed on an
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  * KIND, either express or implied.  See the License for the
+  * specific language governing permissions and limitations
+  * under the License.
+  */
+package org.codehaus.mojo.flatten;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.enterprise.inject.Specializes;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.ModelBase;
+import org.apache.maven.model.building.ModelBuildingRequest;
+import org.apache.maven.model.building.ModelProblemCollector;
+import org.apache.maven.model.inheritance.DefaultInheritanceAssembler;
+import org.apache.maven.model.merge.MavenModelMerger;
+
+/**
+ * The DefaultInheritanceAssembler is encapsulating the InheritanceModelMerger.
+ * The only way to override functionality needed is to define an own InheritanceAssembler 
+ * to provide the needed ModelMerger.
+ * The container is providing the InheritanceAssembler.
+ * This class needs to be configured in the 
+ * src/main/resources/META-INF/sisu/javax.inject.Named to be loaded by the container.
+ *
+ * see issue https://github.com/mojohaus/flatten-maven-plugin/issues/220
+ *
+ * @author kemalsoysal
+ * @author ralfluebeck
+ *
+ */
+@Specializes
```

Comment:
Why it is needed?

Reply:
I remember to have read in the documentation, that the `@Specializes` is needed for a class overriding a base class because of the interface .... I give it a try to remove it and run my tests...

Reply:
removed with commit d2f45da